jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectionProvider.kt

```diff
@@ -188,9 +189,18 @@ class CawsConnectionProvider : GatewayConnectionProvider {
                                     }
                                 }
                             } as ToolkitInstallSettings?
+                            val timeTakenToCheckInstallation = System.currentTimeMillis() - startTimeToCheckInstallation
                             toolkitInstallSettings ?: let {
                                 // environment is non-responsive to SSM; restart
                                 LOG.warn { "Restarting $envId since it appears unresponsive to SSM Run-Command" }
+                                CodecatalystTelemetry.devEnvironmentWorkflowStatistic(
+                                    project = null,
+                                    userId = userId,
+                                    result = TelemetryResult.Failed,
+                                    codecatalystDevEnvironmentWorkflowStep = "ToolkitInstallationSSMCheck",
+                                    codecatalystDevEnvironmentWorkflowError = "Dev env unresponsive to SSM",
```

Comment:
it would be helpful to have something more concrete

Reply:
Tried checking the output from SSM, we don't receive anything noteworthy, the new message noting the cause of the failure as timeout may be useful for us